src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorDefinitionProvider.ts

```diff
@@ -0,0 +1,39 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { RazorLanguageFeatureBase } from './RazorLanguageFeatureBase';
+
+export class RazorDefinitionProvider
+    extends RazorLanguageFeatureBase
+    implements vscode.DefinitionProvider {
+
+    public async provideDefinition(
+        document: vscode.TextDocument, position: vscode.Position,
+        token: vscode.CancellationToken) {
+
+        const projection = await this.getProjection(document, position, token);
+        if (projection) {
```

Comment:
nit: to avoid extra indentation bail early. Aka

```TypeScript
if (!projection) {
    return;
}

...
```

It's super minor but from experience has helped aid in reading quality a ton.